src/driver/op.rs

```diff
@@ -130,30 +148,69 @@ where
                 Poll::Pending
             }
             Lifecycle::Ignored(..) => unreachable!(),
-            Lifecycle::Completed(result, flags) => {
-                inner.ops.remove(me.index);
-                me.index = usize::MAX;
-
-                Poll::Ready(me.data.take().unwrap().complete(result, flags))
+            Lifecycle::Completed(v) => {
+                let data = me.data.take().unwrap();
+                let updated = v
+                    .into_iter()
+                    .fold(Update::More(data), |data, (result, flags)| {
```

Comment:
maybe the variable name 'data' in More(data) is confusing me. Is the More(op) below the same type?

Reply:
Gone now